front/lib/resources/agent_message_feedback_resource.ts

```diff
@@ -43,130 +61,357 @@ export class AgentMessageFeedbackResource extends BaseResource<AgentMessageFeedb
+              as: "message",
+              attributes: ["id", "sId"],
+              include: [
+                {
+                  model: Conversation,
+                  as: "conversation",
+                  attributes: ["id", "sId"],
+                },
+              ],
+            },
+          ],
+        },
+        {
+          model: UserResource.model,
+          attributes: ["name", "imageUrl", "email"],
+        },
+      ],
+      order: [
+        ["agentConfigurationVersion", "DESC"],
+        ["createdAt", "DESC"],
+      ],
+      limit: filters?.limit,
     });
 
-    if (!agentMessageFeedback) {
-      return null;
-    }
+    return (
+      agentMessageFeedback
+        // Typeguard needed because of TypeScript limitations
+        .filter(
+          (
+            feedback
+          ): feedback is AgentMessageFeedback & {
+            agentMessage: { message: Message & { conversation: Conversation } };
+          } => !!feedback.agentMessage?.message?.conversation
+        )
+        .map((feedback) => {
+          return {
+            id: feedback.id,
+            messageId: feedback.agentMessage.message.sId,
+            agentMessageId: feedback.agentMessageId,
+            userId: feedback.userId,
+            thumbDirection: feedback.thumbDirection,
+            content: feedback.content
+              ? feedback.content.replace(/\r?\n/g, "\\n")
+              : null,
+            isConversationShared: feedback.isConversationShared,
+            createdAt: feedback.createdAt,
+            agentConfigurationId: feedback.agentConfigurationId,
+            agentConfigurationVersion: feedback.agentConfigurationVersion,
 
-    return new AgentMessageFeedbackResource(
-      AgentMessageFeedback,
-      agentMessageFeedback.get()
+            ...(withMetadata && {
+              // This field is sensitive, it allows accessing the conversation
+              conversationId: feedback.isConversationShared
+                ? feedback.agentMessage.message.conversation.sId
+                : null,
+              userName: feedback.user.name,
+              userEmail: feedback.user.email,
+              userImageUrl: feedback.user.imageUrl,
+            }),
+          };
+        })
     );
   }
 
-  static async fetchByUserAndAgentMessages(
-    user: UserType,
-    agentMessages: AgentMessage[]
-  ): Promise<AgentMessageFeedbackResource[]> {
-    const agentMessageFeedback = await AgentMessageFeedback.findAll({
+  static async getConversationFeedbacksForUser(
+    auth: Authenticator,
+    conversation: ConversationType | ConversationWithoutContentType
+  ): Promise<Result<AgentMessageFeedbackType[], ConversationError | Error>> {
+    const owner = auth.workspace();
+    if (!owner) {
+      return new Err(new Error("workspace_not_found"));
+    }
+    const user = auth.user();
+    if (!user) {
+      return new Err(new Error("user_not_found"));
+    }
+
+    const feedbackForMessages = await Message.findAll({
```

Comment:
Did you explain analyze this one?

Reply:
I didn't, conversation.ts has similar ones (e.g. getConversation) that seem more expensive (more include clauses, and ranking), so it is unlikely to be a problem

Reply:
Let's check nonetheless

Reply:
Tested on a conv with some feedbacks and >10 messages

`-> Nested Loop (cost=4.73..38.01 rows=3 width=381) (actual time=0.023..0.060 rows=6 loops=1)
-> Bitmap Heap Scan on agent_message_feedbacks "agentMessage->feedbacks" (cost=4.30..12.66 rows=3 width=75) (actual time=0.013..0.017 rows=6 loops=1)
Recheck Cond: ("userId" = 58255)
Heap Blocks: exact=3
-> Bitmap Index Scan on "agent_message_feedbacks_userId_fk" (cost=0.00..4.30 rows=3 width=0) (actual time=0.007..0.007 rows=6 loops=1)
Index Cond: ("userId" = 58255)
-> Index Scan using agent_messages_pkey on agent_messages "agentMessage" (cost=0.43..8.45 rows=1 width=306) (actual time=0.006..0.006 rows=1 loops=6)
Index Cond: (id = "agentMessage->feedbacks"."agentMessageId")
Planning Time: 0.402 ms
Execution Time: 0.176 ms`